src/job.ts

```diff
@@ -437,36 +443,35 @@ export class Job {
             writeStreams.stdout(chalk`${this.getJobNameString()} {magentaBright copied artifacts to cwd} in {magenta ${prettyHrtime(endTime)}}\n`);
         }
 
-        const cp = childProcess.spawn(this.containerId ? `docker start --attach -i ${this.containerId}` : "bash -e", {
-            shell: "bash",
-            stdio: ["pipe", "pipe", "pipe"],
-            cwd: this.cwd,
-        });
-
-        cp.stdin.write("set -eo pipefail\n");
+        let cmd = "set -eo pipefail\n";
+        cmd += "exec 0< /dev/null\n";
 
         if (this.imageName) {
```

Comment:
Maybe introduce imageJob instead of all the if this.imageName

Reply:
You talking about a JobBase.ts, ShellJob.ts and DockerJob.ts pattern? @moberghammer

Reply:
Yer something like that so you only need a single if

Reply:
Yay, its a lot of work for this. But its a good idea. Won't do it in this PR.